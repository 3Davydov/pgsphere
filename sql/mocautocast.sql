-- These are MOCs generated by
-- SELECT smoc(mocorder, scircle '<(27d, -43d), 0.1d>');
-- intended to check automatic casts of other geometries.

CREATE TABLE varorders (
	mocorder smallint,
	geo smoc);
INSERT INTO varorders (mocorder, geo) VALUES
	(1, smoc('1/32 34 1/')),
	(-3, smoc('1/32 34 3/')),
	(3, smoc('3/547-548 550 553 3/')),
	(7, smoc('7/140857-140860 140862 140944-140945 7/')),
	(-7, smoc('1/32 34 7/'));

-- OVERLAPS smoc/scircle

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gsupc
FROM varorders 
WHERE geo @> scircle '<(30d, -43d), 1d>';

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gnsupc
FROM varorders 
WHERE geo !@> scircle '<(30d, -43d), 1d>';

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) csubg
FROM varorders 
WHERE scircle '<(30d, -43d), 1d>' <@ geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cnsubg
FROM varorders 
WHERE scircle '<(30d, -43d), 1d>' !<@ geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gsubc
FROM varorders 
WHERE geo <@ scircle '<(30d, -43d), 1d>';

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gnsubc
FROM varorders 
WHERE geo !<@ scircle '<(30d, -43d), 1d>';

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) csupg
FROM varorders 
WHERE scircle '<(30d, -43d), 1d>' @> geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cnsupg
FROM varorders 
WHERE scircle '<(30d, -43d), 1d>' !@> geo;


-- OVERLAPS smoc/spoly

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gsupp
FROM varorders 
WHERE geo @> spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}');

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gnsupp
FROM varorders 
WHERE geo !@> spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}');

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) psubg
FROM varorders 
WHERE spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}') <@ geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) pnsubg
FROM varorders 
WHERE spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}') !<@ geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gsubp
FROM varorders 
WHERE geo <@ spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}');

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gnsubp
FROM varorders 
WHERE geo !<@ spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}');

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) psupg
FROM varorders 
WHERE spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}') @> geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) pnsupg
FROM varorders 
WHERE spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}') !@> geo;


-- INTERSECTS smoc/scircle

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cgim
FROM varorders 
WHERE scircle '<(45d, -40d), 1d>' && geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cgnim
FROM varorders 
WHERE scircle '<(45d, -40d), 1d>' !&& geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cmig
FROM varorders 
WHERE geo && scircle '<(45d, -40d), 1d>';

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cnim
FROM varorders 
WHERE geo !&& scircle '<(45d, -40d), 1d>';


-- INTERSECTS smoc/spoly

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cgim
FROM varorders 
WHERE spoly '{(51.1d, -50.1d), (45.8d, -47.5), (46.7d, -43.7d)}' && geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cgnim
FROM varorders 
WHERE spoly '{(51.1d, -50.1d), (45.8d, -47.5), (46.7d, -43.7d)}' !&& geo;

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cmig
FROM varorders 
WHERE geo && spoly '{(51.1d, -50.1d), (45.8d, -47.5), (46.7d, -43.7d)}';

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cnim
FROM varorders 
WHERE geo !&& spoly '{(51.1d, -50.1d), (45.8d, -47.5), (46.7d, -43.7d)}';


DROP TABLE varorders;
