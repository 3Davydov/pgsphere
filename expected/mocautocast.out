-- These are MOCs generated by
-- SELECT smoc(mocorder, scircle '<(27d, -43d), 0.1d>');
-- intended to check automatic casts of other geometries.
CREATE TABLE varorders (
	mocorder smallint,
	geo smoc);
INSERT INTO varorders (mocorder, geo) VALUES
	(1, smoc('1/32 34 1/')),
	(-3, smoc('1/32 34 3/')),
	(3, smoc('3/547-548 550 553 3/')),
	(7, smoc('7/140857-140860 140862 140944-140945 7/')),
	(-7, smoc('1/32 34 7/'));
-- OVERLAPS smoc/scircle
SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gsupc
FROM varorders 
WHERE geo @> scircle '<(30d, -43d), 1d>';
 gsupc 
-------
 -7/-3
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gnsupc
FROM varorders 
WHERE geo !@> scircle '<(30d, -43d), 1d>';
  gnsupc  
----------
  1/ 3/ 7
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) csubg
FROM varorders 
WHERE scircle '<(30d, -43d), 1d>' <@ geo;
 csubg 
-------
 -7/-3
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cnsubg
FROM varorders 
WHERE scircle '<(30d, -43d), 1d>' !<@ geo;
  cnsubg  
----------
  1/ 3/ 7
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gsubc
FROM varorders 
WHERE geo <@ scircle '<(30d, -43d), 1d>';
 gsubc 
-------
  1
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gnsubc
FROM varorders 
WHERE geo !<@ scircle '<(30d, -43d), 1d>';
   gnsubc    
-------------
 -7/-3/ 3/ 7
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) csupg
FROM varorders 
WHERE scircle '<(30d, -43d), 1d>' @> geo;
 csupg 
-------
  1
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cnsupg
FROM varorders 
WHERE scircle '<(30d, -43d), 1d>' !@> geo;
   cnsupg    
-------------
 -7/-3/ 3/ 7
(1 row)

-- OVERLAPS smoc/spoly
SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gsupp
FROM varorders 
WHERE geo @> spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}');
 gsupp 
-------
 -7/-3
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gnsupp
FROM varorders 
WHERE geo !@> spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}');
  gnsupp  
----------
  1/ 3/ 7
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) psubg
FROM varorders 
WHERE spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}') <@ geo;
 psubg 
-------
 -7/-3
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) pnsubg
FROM varorders 
WHERE spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}') !<@ geo;
  pnsubg  
----------
  1/ 3/ 7
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gsubp
FROM varorders 
WHERE geo <@ spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}');
 gsubp 
-------
  1
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) gnsubp
FROM varorders 
WHERE geo !<@ spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}');
   gnsubp    
-------------
 -7/-3/ 3/ 7
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) psupg
FROM varorders 
WHERE spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}') @> geo;
 psupg 
-------
  1
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) pnsupg
FROM varorders 
WHERE spoly('{(26d,-42d), (26d,-41d), (27d,-41d)}') !@> geo;
   pnsupg    
-------------
 -7/-3/ 3/ 7
(1 row)

-- INTERSECTS smoc/scircle
SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cgim
FROM varorders 
WHERE scircle '<(45d, -40d), 1d>' && geo;
   cgim   
----------
 -7/-3/ 1
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cgnim
FROM varorders 
WHERE scircle '<(45d, -40d), 1d>' !&& geo;
 cgnim 
-------
  3/ 7
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cmig
FROM varorders 
WHERE geo && scircle '<(45d, -40d), 1d>';
   cmig   
----------
 -7/-3/ 1
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cnim
FROM varorders 
WHERE geo !&& scircle '<(45d, -40d), 1d>';
 cnim  
-------
  3/ 7
(1 row)

-- INTERSECTS smoc/spoly
SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cgim
FROM varorders 
WHERE spoly '{(51.1d, -50.1d), (45.8d, -47.5), (46.7d, -43.7d)}' && geo;
   cgim   
----------
 -7/-3/ 1
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cgnim
FROM varorders 
WHERE spoly '{(51.1d, -50.1d), (45.8d, -47.5), (46.7d, -43.7d)}' !&& geo;
 cgnim 
-------
  3/ 7
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cmig
FROM varorders 
WHERE geo && spoly '{(51.1d, -50.1d), (45.8d, -47.5), (46.7d, -43.7d)}';
   cmig   
----------
 -7/-3/ 1
(1 row)

SELECT string_agg(to_char(mocorder, '9'), '/' ORDER BY mocorder) cnim
FROM varorders 
WHERE geo !&& spoly '{(51.1d, -50.1d), (45.8d, -47.5), (46.7d, -43.7d)}';
 cnim  
-------
  3/ 7
(1 row)

DROP TABLE varorders;
